client/src/components/Production/ProductionModal.jsx

```diff
@@ -0,0 +1,151 @@
+import ProductionModalBike from 'components/Production/ProductionModalBike.jsx';
+
+const ProductionModal = () => {
+  const {
+    formStyle,
+    elementStyle,
+    isOpen,
+    onOpen,
+    onClose,
+    form,
+    setForm,
+    colorRef,
+    quantityRef,
+    statusRef,
+    nameRef,
+    descriptionRef,
+    startDate,
+    endDate,
+    assemblyMachineRef,
+    noteRef,
+    profitMarginRef,
+    partRefs,
+    bikeRefs,
+    handleStartDateInput,
+    handleEndDateInput,
+    handleSubmit,
+  } = useProductionModal();
+
+  return (
+    <>
+      <Center mt={4}>
+        <Button onClick={onOpen}>Produce</Button>
+      </Center>
+      <Modal isOpen={isOpen} onClose={onClose}>
+        <ModalOverlay />
+        <ModalContent>
+          <ModalHeader>Production</ModalHeader>
+          <ModalCloseButton />
+          <ModalBody pb={6}>
+            <Button mr={3} onClick={() => setForm('Part')}>
+              Part
+            </Button>
+            <Button onClick={() => setForm('Bike')}>Bike</Button>
+            <div style={formStyle}>
+              <FormControl isRequired style={elementStyle}>
+                <FormLabel>Status</FormLabel>
+                <Select ref={statusRef}>
+                  <Fragment>
+                    <option>Ongoing</option>
+                    <option>Complete</option>
+                    <option>Idle</option>
+                    <option>Cancelled</option>
+                  </Fragment>
+                </Select>
+              </FormControl>
+              <FormControl isRequired style={elementStyle}>
+                <FormLabel>Name</FormLabel>
+                <Input ref={nameRef} />
+              </FormControl>
+              <FormControl isRequired style={elementStyle}>
+                <FormLabel>Description</FormLabel>
+                <Input ref={descriptionRef} />
+              </FormControl>
+            </div>
+            {form === 'Part' ? (
+              <ProductionModalPart ref={partRefs} />
+            ) : (
+              <ProductionModalBike ref={bikeRefs} />
+            )}
+            <div style={formStyle}>
+              <FormControl isRequired style={elementStyle}>
+                <FormLabel>Color</FormLabel>
+                <Input ref={colorRef} />
+              </FormControl>
+              <FormControl isRequired style={elementStyle}>
+                <FormLabel>Quantity</FormLabel>
+                <NumberInput min={1} defaultValue={1}>
+                  <NumberInputField ref={quantityRef} />
+                  <NumberInputStepper>
+                    <NumberIncrementStepper />
+                    <NumberDecrementStepper />
+                  </NumberInputStepper>
+                </NumberInput>
+              </FormControl>
+              <FormControl isRequired style={elementStyle}>
+                <FormLabel>Profit Margin</FormLabel>
+                <NumberInput min={1.1} defaultValue={1.1}>
```

Comment:
if you try to set profit margin to a float, it throws a http code status 400 because in the API you expect an integer

Reply:
Yeah you need to clone and test locally, Siu made some changes to the backend in this PR

Reply:
Yea, I thought I changed it in the end point PR, but I guess it was in this one...